src/repository/opamRepository.ml

```diff
@@ -396,10 +396,12 @@ let apply_repo_update repo = function
     Done ()
   | Update_err _ -> assert false
 
-let cleanup_repo_update = function
-  | Update_full d -> OpamFilename.rmdir d
-  | Update_patch f -> OpamFilename.remove f
-  | _ -> ()
+let cleanup_repo_update upd =
+  if not (OpamConsole.debug ()) then
+    match upd with
+    | Update_full d -> OpamFilename.rmdir d
+    | Update_patch f -> OpamFilename.remove f
+    | _ -> ()
```

Comment:
It was a git repo I was testing this against - I'll do some further testing with several updates in a row.

One related problem I did hit with the patch failing (because of the CRLF problems) was that it leaves .orig and .rej files in the git repo clone - I don't know whether it would be worth the backend always doing a `git clean -dfx`?